src/systems/filecoin_markets/storage_market/storage_client.id

```diff
@@ -1,18 +1,62 @@
 import ipld "github.com/filecoin-project/specs/libraries/ipld"
-import deal "github.com/filecoin-project/specs/systems/filecoin_markets/deal"
-import piece "github.com/filecoin-project/specs/systems/filecoin_files/piece"
+import deal "github.com/filecoin-project/specs/systems/filecoin_markets/storage_market/storage_deal"
+import actor "github.com/filecoin-project/specs/systems/filecoin_vm/actor"
+import libp2p "github.com/filecoin-project/specs/libraries/libp2p"
+import addr "github.com/filecoin-project/specs/systems/filecoin_vm/actor/address"
+import block "github.com/filecoin-project/specs/systems/filecoin_blockchain/struct/block"
+
+type ClientDeal struct {
+    ProposalCid     ipld.CID
+    Proposal        deal.StorageDealProposal
+    State           StorageDealStatus
+    Miner           libp2p.PeerID
+    MinerWorker     addr.Address
+    DealID          deal.DealID
+    PublishMessage  ipld.CID
+}
+
+type ProposeStorageDealResult struct {
+    ProposalCid ipld.CID
+}
+
+type StorageProviderInfo struct {
+    Address     addr.Address  // actor address
+    Owner       addr.Address
+    Worker      addr.Address  // signs messages
+    SectorSize  uint64
+    PeerID      libp2p.PeerID
+}
 
 type StorageClient struct {
-    // generate PieceCID(CommP) from payload
-    // so that client can identify their deal
-    generatePieceCID(payloadCID ipld.CID) piece.PieceCID
+    // ListProviders queries chain state and returns active storage providers
+    ListProviders()        (providers [StorageProviderInfo], error)
+
+    // ListDeals lists on-chain deals associated with this client
+    ListDeals()            (deals [deal.StorageDeal], error)
+
+    // ListInProgressDeals lists deals that are in progress or rejected
+    // - but not on chain
```

Comment:
why or rejected? that sounds like those are not in progress..?

Reply:
a rejected deal is not on chain -- this should be "deals not on chain" -- the only record of the rejection is in the clients local store.

Reply:
I think I will rename this ListLocalDeals